File: api/travel-agent.js
import { askTravelAgent } from "../scripts/travel-agent.mjs";

export default async function handler(request, response) {
  response.setHeader("Access-Control-Allow-Origin", "*");
  response.setHeader("Access-Control-Allow-Methods", "POST, OPTIONS");
  response.setHeader("Access-Control-Allow-Headers", "content-type");

  if (request.method === "OPTIONS") {
    response.status(204).end();
    return;
  }

  if (request.method !== "POST") {
    response.status(405).json({ error: "Method not allowed" });
    return;
  }

  try {
    const result = await askTravelAgent(request.body || {});
    response.status(200).json(result);
  } catch (error) {
    response.status(502).json({
      error: "LangChain Agent unavailable",
      detail: error.message,
    });
  }
}

File: api/travel-agent.py
from http.server import BaseHTTPRequestHandler

from python_agent_backend import build_travel_agent_response, parse_json_body


class handler(BaseHTTPRequestHandler):
    def _set_headers(self, status_code: int, content_type: str = "application/json; charset=utf-8") -> None:
        self.send_response(status_code)
        self.send_header("Access-Control-Allow-Origin", "*")
        self.send_header("Access-Control-Allow-Methods", "POST, OPTIONS")
        self.send_header("Access-Control-Allow-Headers", "content-type")
        self.send_header("Content-Type", content_type)
        self.end_headers()

    def do_OPTIONS(self) -> None:
        self._set_headers(204)

    def do_POST(self) -> None:
        try:
            body = parse_json_body(self)
            payload = build_travel_agent_response(body)
            self._set_headers(200)
            self.wfile.write(__import__("json").dumps(payload, ensure_ascii=False).encode("utf-8"))
        except ValueError as error:
            self._set_headers(400)
            self.wfile.write(__import__("json").dumps({"error": str(error)}, ensure_ascii=False).encode("utf-8"))
        except Exception as error:  # pragma: no cover - Vercel runtime path
            self._set_headers(502)
            self.wfile.write(__import__("json").dumps({
                "error": "Python Agent unavailable",
                "detail": str(error),
            }, ensure_ascii=False).encode("utf-8"))

    def do_GET(self) -> None:
        self._set_headers(405)
        self.wfile.write(b'{"error":"Method not allowed"}')
